src/common/snippets/src/lowered/pass/optimize_loop_single_evaluation.cpp

```diff
@@ -16,30 +17,35 @@ namespace pass {
 
 bool OptimizeLoopSingleEvaluation::run(lowered::LinearIR& linear_ir, lowered::LinearIR::constExprIt begin, lowered::LinearIR::constExprIt end) {
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::OptimizeLoopSingleEvaluation")
+    const auto& loop_manager = linear_ir.get_loop_manager();
+
     bool is_modified = false;
     for (auto expr_it = begin; expr_it != end; ++expr_it) {
         const auto& expr = *expr_it;
         if (auto loop_end = ov::as_type_ptr<op::LoopEnd>(expr->get_node())) {
-            // *1* solo vector/tail loop + empty outer loop
-            //      => skip increments (both counter & ptr) : set evaluate_once flag
-            // *2* solo vector/tail loop + non-empty outer loop
-            //      => skip counter increments but perform ptr increments : set evaluate_once,
-            //         and perform pointer increments through finalization offsets
-            // *3* vector loop(s) + one tail loop
-            //      => vector as usual, tail depends on outer loop, see *1* and *2*
-            if (loop_end->has_dynamic_params() || loop_end->get_work_amount() >= 2 * loop_end->get_increment())
-                continue;
-
-            auto new_finalization_offsets = loop_end->get_finalization_offsets();
-            const auto& ptr_increments = loop_end->get_ptr_increments();
-            const auto work_amount_incr = static_cast<int64_t>(loop_end->get_increment());
-            for (size_t i = 0; i < new_finalization_offsets.size(); i++) {
-                new_finalization_offsets[i] += ptr_increments[i] * work_amount_incr;
+            const auto& loop_info = loop_manager->get_loop_info<ExpandedLoopInfo>(loop_end->get_id());
+            const auto work_amount = loop_end->get_work_amount();
+            const auto increment = loop_end->get_increment();
+
+            if ((!utils::is_dynamic_value(work_amount) && work_amount == increment) || (loop_info->is_evaluate_once())) {
```

Comment:
1. Because this condition is outdated. This pass is called after [InsertSpecificIterations](https://github.com/openvinotoolkit/openvino/blob/3f47ab6d73568694b8e7ac793ecb50096be46c19/src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp#L78) where we always set new `work_amount` which is divided entirely by `increment`. Moreover, your example is unreal now because of new calculations of `work_amount` and `increment` in `InsertSpecificIterations`

Reply:
2. I think it can be resolve so: some passes can manually set `evaluate_once` (for example, brgemm blocking related passes). Automatically `evaluate_once` can be updated in `InsertSpecificIterations` if `work_amount == increment`. Then we call `OptimizeLoopSingleEvaluation` which checks `LoopInfo->is_evaluate_once()` and if it's true, update `LoopInfo` and `LoopEnd.
I implemented so now, could you please review one more time?